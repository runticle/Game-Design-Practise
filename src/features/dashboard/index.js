import React from 'react'

function Dashboard(props) {
  //
  // store.dispatch({ type: 'ADD_TILES', payload: {
  //   tiles: store.getState().map.tiles,
  // }})

  return (
    <div
      style = {{
        border: '4px solid white',
        top: '100px',
        left: '5px',
        position: 'relative',
        width: '800px',
        height: '50px',
        margin: '0 auto',
        backgroundColor: 'brown'
      }}
    >
    </div>
  )
}

export default Dashboard
