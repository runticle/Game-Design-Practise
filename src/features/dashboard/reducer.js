const initialState = {
  messageIndex: 0,
  inventory: [],
  rubies: 0
}

const dashboardReducer = (state = initialState, action ) => {
  switch(action.type) {
    case 'SET_MESSAGE':
        const newMsgIndex = action.payload.messageIndex
        return { ...state, messageIndex: newMsgIndex }
    case 'SEND_TO_INVENTORY':
        const newInv = action.payload.inventory
        return { ...state, inventory: state.inventory.concat(newInv) }
    case 'ADD_RUBY':
        return { ...state, rubies: state.rubies + 1 }
    default:
      return state
  }
}

export default dashboardReducer
