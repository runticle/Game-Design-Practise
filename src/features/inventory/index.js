import React from 'react'
import { SPRITE_SIZE } from '../../config/constants'

function getItemSprite(type) {
  switch(type) {
    case 2:
      return 'dragon-egg'
    case 6:
      return 'tree'
    case 10:
      return 'axe'
    case 11:
      return 'ruby'
    default:
      return null
  }
}

function Item(props) {
  return (
    <div
      className={`item ${getItemSprite(props.item)}`}
      style = {{
        height: SPRITE_SIZE,
        width: SPRITE_SIZE,
        float: 'left'
      }}
    ></div>
  )
}

function Inventory(props) {

  return (
    <div
      id={'inventory'}
      style = {{
        float: 'left',
        color: 'white',
        width: '67%',
        height: '80%',
        padding: '5px',
        overflow: 'hidden',
        backgroundColor: "black"
    }}
    >
      {
        props.inventory.map(item => <Item item={item} />)
      }
    </div>
  )

}

export default Inventory
