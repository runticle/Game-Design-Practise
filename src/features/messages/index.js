import React from 'react'

export function Messages(props) {

  return (
    <div style = {{
      float: 'right',
      color: 'white',
      width: '30%',
      height: '80%',
      padding: '5px',
      backgroundColor: "grey"
    }}>
      {props.message}
    </div>
  )

}

export default Messages
