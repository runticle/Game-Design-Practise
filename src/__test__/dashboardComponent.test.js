import React from 'react'
import { shallow } from 'enzyme'

import {Dashboard, Rubies} from '../features/dashboard/index'
import { Inventory } from '../features/inventory/index'
import {Messages} from '../features/messages/index'

describe('Dashboard', () => {
  describe('Dashboard', () => {

    it('should render correctly', () => {
      let props
      const dashboard = shallow(<Dashboard props={props}/>)

      expect(dashboard.find('div').length).toEqual(1)
      expect(dashboard).toMatchSnapshot()
    })

    it('should render an Inventory', () => {
      let props
      const dashboard = shallow(<Dashboard props={props}/>)

      expect(dashboard.containsMatchingElement(<Inventory />)).toEqual(true);
    })

    it('should render Rubies', () => {
      let props
      const dashboard = shallow(<Dashboard props={props}/>)

      expect(dashboard.containsMatchingElement(<Rubies />)).toEqual(true);
    })

    it('should render Messages', () => {
      let props
      const dashboard = shallow(<Dashboard props={props}/>)

      expect(dashboard.containsMatchingElement(<Messages />)).toEqual(true);
    })

  })
})
