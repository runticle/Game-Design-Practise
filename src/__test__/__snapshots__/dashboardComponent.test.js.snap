// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Dashboard should render correctly 1`] = `
<div
  className="dashboard"
  style={
    Object {
      "border": "4px solid white",
      "height": "50px",
      "left": "5px",
      "margin": "0 auto",
      "padding": "4px",
      "position": "relative",
      "top": "100px",
      "width": "800px",
    }
  }
>
  <Rubies />
  <Inventory />
  <Messages />
</div>
`;
